ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -158,10 +162,28 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
     lifecycleScope.launch { dataCollectionViewModel.onPreviousClicked(viewModel) }
   }
 
-  fun moveToNext() {
+  private fun moveToNext() {
     lifecycleScope.launch { dataCollectionViewModel.onNextClicked(viewModel) }
   }
 
+  fun handleNext() {
+    if (getTask().isAddLoiTask) {
```

Comment:
how about using `when`, so that if in future, checks increase, we don't have to do lot of nesting.

Reply:
Good call, however, received a linter error: `Prefer using 'if' for binary conditions instead of 'when'. [UseIfInsteadOfWhen]`.